subgraphs/compound-forks/protocols/venus/src/mapping.ts

```diff
@@ -293,6 +342,103 @@ export function handleTransfer(event: Transfer): void {
   );
 }
 
+// update market reward amounts based on event data.
+export function handleVenusSpeedUpdated(event: VenusSpeedUpdated): void {
+  const marketAddress = event.params.vToken;
+  const speed = event.params.newSpeed;
+  const rewards = getRewardsPerDay(
+    event.block.timestamp,
+    event.block.number,
+    speed.toBigDecimal(),
+    RewardIntervalType.BLOCK
+  ).times(BigDecimal.fromString(DAYS_PER_YEAR.toString()));
+
+  const rewardAmount = BigInt.fromString(rewards.truncate(0).toString());
+  const borrowRewardToken = getOrCreateRewardToken(RewardTokenType.BORROW);
+  const supplyRewardToken = getOrCreateRewardToken(RewardTokenType.DEPOSIT);
+
+  const market = Market.load(marketAddress.toHexString())!;
+  market.rewardTokens = [
+    borrowRewardToken.rewardToken.id,
+    supplyRewardToken.rewardToken.id,
+  ];
+  market.rewardTokenEmissionsAmount = [rewardAmount, rewardAmount]; // venus gives the same amount to borrowers and suppliers for each market
+  market.rewardTokenEmissionsUSD = [BIGDECIMAL_ZERO, BIGDECIMAL_ZERO];
+  updateRewardValueUSD(market);
+  market.save();
+
+  // update market snapshots.
+  getOrCreateMarketDailySnapshot(
+    market,
+    event.block.timestamp,
+    event.block.number
+  );
+  getOrCreateMarketHourlySnapshot(
+    market,
+    event.block.timestamp,
+    event.block.number
+  );
+}
+
+// updateRewardValueUSD will update the reward value in USD, assuming
+// the reward amount hasn't changed. If it changes it will be updated on VenusSpeedUpdated.
+// This function can be called anytime and it will update the reward value in USD.
+function updateRewardValueUSD(market: Market): void {
+  if (!market.rewardTokens || market.rewardTokens!.length == 0) {
+    return;
+  }
+
+  const protocol = getOrCreateProtocol();
+  const oracleContract = PriceOracle.bind(
+    Address.fromString(protocol._priceOracle)
+  );
+  const priceCall = oracleContract.try_getUnderlyingPrice(vXVS.address);
+  if (priceCall.reverted) {
+    log.error("unable to calculate rewards: priceCall reverted for vXVS {}", [
```

Comment:
Help me understand why an error makes more sense here

Reply:
I tend to avoid warnings, because of the unclear meaning from being in between an error and an info. If there is some action that should be taken I default to error. If not, info.

This call should not revert I think? If it does, we should maybe get the price from a different source for the blocks where it fails? So an error would inform us about that (if we could have alerts or something).

https://dave.cheney.net/2015/11/05/lets-talk-about-logging

Reply:
makes sense, I think there are differing opinions on logging in software eng in general. But I will think about this moving forward. I usually default to warnings, and don't log anything else

Reply:
>  I usually default to warnings, and don't log anything else

Damn, haha 😅 

The problem there is basically that. It is not clear what we should do about it once a warning pops up somewhere. Is it safe to ignore? Or should we fix it somehow.